lib/Runtime/Library/JavascriptMap.cpp

```diff
@@ -64,6 +64,15 @@ namespace Js
 
         Var iterable = (args.Info.Count > 1) ? args[1] : library->GetUndefined();
 
+        if (mapObject->map != nullptr)
+        {
+            JavascriptError::ThrowTypeErrorVar(scriptContext, JSERR_ObjectIsAlreadyInitialized, _u("Map"), _u("Map"));
+        }
+
+        // ensure mapObject->map is created before trying to fetch the adder function
+        // see github#2747
+        mapObject->map = RecyclerNew(scriptContext->GetRecycler(), MapDataMap, scriptContext->GetRecycler());
```

Comment:
Set was vulnerable to the same issue, I just fixed it there. I can look tonight or tomorrow at other classes that may have the same issue -- I would imagine candidates include any where you can give initial data to the constructor that the corresponding NewInstance function simply calls this->insert() with, especially ES2015 APIs like Set and Map.